qpdf/qpdf-ctest.c

```diff
@@ -530,7 +530,9 @@ static void test24(char const* infile,
     assert(! qpdf_oh_is_scalar(qpdf, mediabox));
     assert(qpdf_oh_is_array(qpdf, mediabox));
     assert(qpdf_oh_get_array_n_items(qpdf, mediabox) == 4);
-    assert(qpdf_oh_wrap_in_array(qpdf, mediabox) == mediabox);
+    qpdf_oh wrapped_mediabox = qpdf_oh_wrap_in_array(qpdf, mediabox);
+    assert(wrapped_mediabox != mediabox);
```

Comment:
We will need to strengthen this test a little to make sure we are getting the same underlying object. Otherwise, this test would pass if qpdf_oh_wrap_in_array did some random crazy thing like returning a null object. I am happy to enhance the test myself. I would do two things:
* For this test, I would check in the for loop that the items in wrapped_mediabox and mediabox match
* I would add a test that wraps an array containing an indirect object and makes sure that the internal object has the same object ID as the original array's internal object.

I'll go ahead and approve CI for this, then I can either code the additional test or you can. It would take about as long to code it as it took me to type this comment, so I don't mind taking care of it if you're not comfortable with it.